src/main/capi/duckdb_value-c.cpp

```diff
@@ -329,3 +329,35 @@ duckdb_value duckdb_get_map_value(duckdb_value value, idx_t index) {
 	auto &child_struct = duckdb::StructValue::GetChildren(child);
 	return WrapValue(new duckdb::Value(child_struct[1]));
 }
+
+idx_t duckdb_get_list_size(duckdb_value value) {
+	if (!value) {
+		return 0;
+	}
+
+	auto val = UnwrapValue(value);
+	if (val.type().id() != LogicalTypeId::LIST) {
```

Comment:
Can this also check for LogicalTypeId::MAP, that should also be allowed

Reply:
@Tishj Updated now.